### ISSUE TITLE
Order of optimizer.step() and lr_scheduler.step()

### ISSUE THREAD

**Issue by TaoXieSZ** [ultralytics/yolov5#313]
Should we be worried about this?
`
/usr/local/lib/python3.6/dist-packages/torch/optim/lr_scheduler.py:114: UserWarning: Seems like `optimizer.step()` has been overridden after learning rate scheduler initialization. Please, make sure to call `optimizer.step()` before `lr_scheduler.step()`. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
`
![image](https://user-images.githubusercontent.com/7379039/86663069-58555b80-bfbb-11ea-984d-170f982d98f3.png)

Thanks!!

**Comment by Laughing-q** [comment]
is this about my fix? [#292 ]  i'm not sure. but i didn't have this warning

**Comment by glenn-jocher** [comment]
I don't see this in Colab Notebook that uses torch 1.5.1. Which environment and torch version are you seeing this in? In any case, it's a very familiar message, it's a pytorch issue that's been around for a while. The worst that happens is the scheduler is ahead or behind by 1 step, not a big deal.

<img width="1435" alt="Screen Shot 2020-07-06 at 7 23 36 PM" src="https://user-images.githubusercontent.com/26833433/86691989-35787680-bfbe-11ea-8a80-88a62d1d38a6.png">

**Comment by TaoXieSZ** [comment]
@glenn-jocher Gotcha! Thanks!
